src/hotspot/os/linux/os_linux.cpp

```diff
@@ -195,10 +195,7 @@ julong os::Linux::available_memory() {
 
   if (OSContainer::is_containerized()) {
     jlong mem_limit, mem_usage;
-    if ((mem_limit = OSContainer::memory_limit_in_bytes()) < 1) {
-      log_debug(os, container)("container memory limit %s: " JLONG_FORMAT ", using host value",
-                             mem_limit == OSCONTAINER_ERROR ? "failed" : "unlimited", mem_limit);
-    }
+    mem_limit = os::physical_memory();
```

Comment:
`os:physical_memory()` will never return a value `<= 0`. Thus, with this change, `OSContainer::memory_usage_in_bytes()` will always be called on a system with no limits set. Previously it (likely) detected that and `mem_limit` was set to `-1` (for unlimited). This is a concern since it would circumvent some caching. `memory_limit_in_bytes` uses caching, `memory_usage_in_bytes` not. I'd suggest to change this to something like this:

```
    jlong mem_limit, mem_usage, host_mem;
    mem_limit = OSContainer::memory_limit_in_bytes();
    host_mem = Linux::physical_memory();
    if (mem_limit > host_mem) {
      mem_limit = -1; // bail out of container code
    }
    [...]
```

`Linux::physical_memory()` is just a field read and `memory_limit_in_bytes()` uses caches. So this should be OK.

Reply:
I see, thank you. Resolved with 60865050954.